### PR TITLE
feat: convert all things to rescript

### PR DESCRIPTION
**Pull request by infothien**
https://github.com/scoville/re-formidable/issues/17

### CODE REVIEW

File: example/App_new.res [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+// }
+
+// // It's pretty easy to have a generic form maker
+// // for when a the same value set and errors are used in different context
+// let makeForm = Formidable.make(~values=module(Values), ~error=module(I18n.Error))
+
+// // In this example, we'll use the hook contained in the module so we need to define it at the top level
+// // In many cases, it's not needed, especially thanks to the Consumer component
+// module Form = (
+//   val makeForm(~onSubmit=_values => (), ~onSubmitError=(_values, _errors) => (), ())
```
Comment (infothien) [inline example/App_new.res]: I tried to covert this file as well but currently I got struggle in this line, and cannot resolve it for now

File: src/Formidable.re [hunk 2]
```diff
@@ -1,636 +0,0 @@
...
-      | _ => None;
-
-    let reset = field => {...field, status: `pristine};
-  };
-
-  let isPristine = fields => fields->Map.String.every(_ => Field.isPristine);
-
-  let isValid = fields => fields->Map.String.every(_ => Field.isValid);
-
-  let isTouched = fields => !fields->Map.String.some(_ => Field.isPristine);
```
Comment (infothien) [inline src/Formidable.re:74]: not sure why here is !fields. Was it a mistake? (I made it as fields in rescript file)

Comment (gaku-sei) [reply src/Formidable.re:74]: You're right, I think it should be `let isTouched = fields => fields->Map.String.some(_ => !Field.isPristine)` in ReScript

File: example/App.res [hunk 3]
```diff
@@ -1,57 +1,49 @@
...
-        ~onSubmitError=(_values, _errors) => (),
-        (),
-      )
-);
+module Form = unpack(makeForm(~onSubmit=_values => (), ~onSubmitError=(_values, _errors) => (), ()))
 
+open Validations
 // Validations can be defined in an other module, and re-used easily
-let requiredValidations = Validations.[(`onChange, required)];
+let requiredValidations = list{(#onChange, required)}
```
Comment (gaku-sei) [inline example/App.res:32]: How complicated would it be to use arrays for the validations in the example?

**Review by gaku-sei — APPROVED** [review]
That's really good 😄 

We can see how to improve some parts of the code later, like the `list/array` thing

File: src/Formidable.res [hunk 4]
```diff
@@ -0,0 +1,565 @@
...
+      }
+
+    let reset = field => {...field, status: #pristine}
+  }
+
+  let isPristine = fields => fields->Map.String.every(_ => Field.isPristine)
+
+  let isValid = fields => fields->Map.String.every(_ => Field.isValid)
+
+  let isTouched = fields => fields->Map.String.some(_ => Field.isPristine)
```
Comment (gaku-sei) [inline src/Formidable.res]: `let isTouched = fields => fields->Map.String.some(_ => !Field.isPristine)`
Maybe?

Like "it's touched if _some_ field is _not_ pristine anymore?"

Comment (infothien) [reply src/Formidable.res]: this is true ! I will correct it thank you @gaku-sei -san

Comment (infothien) [reply src/Formidable.res]: oh maybe it should be 
```
  let isTouched = fields => fields->Map.String.some(_ => Field.isTouched)

```
right @gaku-sei -san

Comment (gaku-sei) [reply src/Formidable.res]: Looks good 👍

**Pull request merged**